inlong-sort/sort-connectors/hive/src/main/java/org/apache/inlong/sort/hive/filesystem/AbstractStreamingWriter.java

```diff
@@ -0,0 +1,220 @@
+
+    private static final long serialVersionUID = 1L;
+
+    // ------------------------ configuration fields --------------------------
+
+    private final long bucketCheckInterval;
+
+    private final StreamingFileSink.BucketsBuilder<
+            IN, String, ? extends StreamingFileSink.BucketsBuilder<IN, String, ?>>
+            bucketsBuilder;
+
+    @Nullable
+    private String inLongMetric;
+
+    @Nullable
+    private String auditHostAndPorts;
+
+    // --------------------------- runtime fields -----------------------------
+
+    private transient Buckets<IN, String> buckets;
+
+    private transient StreamingFileSinkHelper<IN> helper;
+
+    private transient long currentWatermark;
+
+    @Nullable
+    private transient SinkMetricData metricData;
+
+    private transient String inLongGroupId;
+
+    private transient String inLongStreamId;
+
+    public AbstractStreamingWriter(
+            long bucketCheckInterval,
+            StreamingFileSink.BucketsBuilder<
+                    IN, String, ? extends StreamingFileSink.BucketsBuilder<IN, String, ?>>
+                    bucketsBuilder,
+            String inLongMetric,
+            String auditHostAndPorts) {
+        this.bucketCheckInterval = bucketCheckInterval;
+        this.bucketsBuilder = bucketsBuilder;
+        this.inLongMetric = inLongMetric;
+        this.auditHostAndPorts = auditHostAndPorts;
+        setChainingStrategy(ChainingStrategy.ALWAYS);
+    }
+
+    /** Notifies a partition created. */
+    protected abstract void partitionCreated(String partition);
+
+    /**
+     * Notifies a partition become inactive. A partition becomes inactive after all the records
+     * received so far have been committed.
+     */
+    protected abstract void partitionInactive(String partition);
+
+    /**
+     * Notifies a new file has been opened.
+     *
+     * <p>Note that this does not mean that the file has been created in the file system. It is only
+     * created logically and the actual file will be generated after it is committed.
+     */
+    protected abstract void onPartFileOpened(String partition, Path newPath);
+
+    /** Commit up to this checkpoint id. */
+    protected void commitUpToCheckpoint(long checkpointId) throws Exception {
+        helper.commitUpToCheckpoint(checkpointId);
+    }
+
+    @Override
+    public void open() throws Exception {
+        super.open();
+        if (inLongMetric != null) {
+            String[] inLongMetricArray = inLongMetric.split(DELIMITER);
+            inLongGroupId = inLongMetricArray[0];
+            inLongStreamId = inLongMetricArray[1];
+            String nodeId = inLongMetricArray[2];
+            metricData = new SinkMetricData(
+                    inLongGroupId, inLongStreamId, nodeId, getRuntimeContext().getMetricGroup(), auditHostAndPorts);
+            metricData.registerMetricsForDirtyBytes(new ThreadSafeCounter());
+            metricData.registerMetricsForDirtyRecords(new ThreadSafeCounter());

```

Comment:
where to compute dirtyBytes and compute dirtyRecords

Reply:
Here in sink node computing dirtyRecords is meaningless. Beacuse any data in memroy here must be able to serialize successfully(dirty data what  I understand is cannot serialize successfully)，other 
 exception like IOException is not the cause of data is dirty, so should not catch them and inc dirty metric counter.